src/profile.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include "iobuf.h"
+
+#if MG_ENABLE_PROFILE
+struct mg_profitem {
+  const char *name;
+  uint32_t timestamp;
+};
+
+#define MG_PROF_ADD(c, name_)                                          \
+  do {                                                                 \
+    struct mg_iobuf *io = &c->prof;                                    \
+    struct mg_profitem *first = (struct mg_profitem *) io->buf;        \
+    struct mg_profitem item = {name_, (uint32_t) mg_millis()};         \
+    if (io->len >= sizeof(*first)) item.timestamp -= first->timestamp; \
+    mg_iobuf_add(io, io->len, &item, sizeof(item));                    \
+  } while (0)
+
+#define MG_PROF_INIT(c)                \
+  do {                                 \
+    mg_iobuf_init(&(c)->prof, 0, 256); \
+    MG_PROF_ADD((c), "init");          \
```

Comment:
Perhaps we could have a special treatment for the "init" case and save that check that is always true for the others ?
Just trying to spare a picosecond... but maybe you have a bright idea

Reply:
Done!